cognite/client/_api/files.py

```diff
@@ -437,12 +437,35 @@ def upload(
                     for file in files:
                         file_path = os.path.join(root, file)
                         basename = os.path.basename(file_path)
-                        tasks.append((FileMetadata(name=basename), file_path, overwrite))
+                        file_metadata = FileMetadata(
+                            name=basename,
+                            source=source,
```

Comment:
seems like these 6 lines appear trice in this function 
can we copy initial `file_metadata` and change `name` only?
```
source=source,
mime_type=mime_type,
metadata=metadata,
asset_ids=asset_ids,
source_created_time=source_created_time,
source_modified_time=source_modified_time,
```

Reply:
sure, I suppose that would lower the risk for such bugs in the future if fields are added